massa-network-worker/src/node_worker.rs

```diff
@@ -410,6 +333,168 @@ impl NodeWorker {
             }
         }
 
+        // 3- Stop node_reader_handle
+        if !reader_joined {
+            // Abort the task otherwise socket_reader.next() is stuck, waiting for some data to read
+            debug!(
+                "node_worker.run_loop.cleanup.node_reader_handle.abort, node_id: {}",
+                self.node_id
+            );
+            node_reader_handle.abort();
+        }
+
```

Comment:
also stop the writer task

Reply:
At this point, the node_writer_handle is already stopped but to (at line 320):

`match node_writer_handle.await {
`
 Trying to abort the node_writer_handle result in a compilation error:

>    --> massa-network-worker/src/node_worker.rs:344:13
>     |
> 131 |         tokio::pin!(node_writer_handle);
>     |         ------------------------------- move occurs because `node_writer_handle` has type `Pin<&mut tokio::task::JoinHandle<Result<(), NetworkError>>>`, which does not implement the `Copy` trait
> ...
> 320 |             match node_writer_handle.await {
>     |                                     ------ `node_writer_handle` moved due to this method call
> ...
> 344 |             node_writer_handle.abort();
>     |             ^^^^^^^^^^^^^^^^^^^^^^^^^^ value borrowed here after move
>     |
>

Reply:
what's different compared to node_reader_handle ? I think we should handle their closing the same way

Reply:
node_writer_handle is mostly waiting writer_command_rx and it will exit when writer_command_tx is dropped (which is done at line 311).

On the other end, node_reader_handle is mostly waiting for socket_reader; the only 'easy' to close this task is to abort it (done at line 338). If you prefer we can investigate, a more elegant solution (like using tokio Notify + select in node_reader_handle)?

Note that the code is always closing first the writer (by first checking writer_joined at line 314) then is always closing the reader (by first checking reader_joined).